toolbox/src/base/Signal.cpp

```diff
@@ -223,19 +277,19 @@ namespace wbt {
             case PortDataTypeUInt32:
             {
                 dataSize = sizeof(uint32_t);
-                address = ((uint32_t*)data) + startIndex;
+                address = data + startIndex;
                 break;
             }
             case PortDataTypeUInt16:
             {
                 dataSize = sizeof(uint16_t);
-                address = ((uint16_t*)data) + startIndex;
+                address = data + startIndex;
```

Comment:
Why did you drop the cast?

Reply:
As discussed f2f, there is currently a bug of the memory management of all signal types excluding `double`. I will refactor `Signal` and `Data` classes with templates and std objects in a second stage.